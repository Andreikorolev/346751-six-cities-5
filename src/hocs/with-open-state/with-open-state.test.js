import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withOpenState from "./with-open-state";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentHOC = withOpenState(MockComponent);

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

describe(`withOpenState render`, () => {
  it(`shoul render correctly`, () => {
    const tree = renderer.create(
        <MockComponentHOC>
          <React.Fragment/>
        </MockComponentHOC>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
