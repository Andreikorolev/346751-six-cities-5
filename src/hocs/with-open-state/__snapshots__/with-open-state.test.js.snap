// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOpenState render shoul render correctly 1`] = `<div />`;
